### PR TITLE
change NestedLoopConsumer to work on a TwoTableJoin relation

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

File: sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                    rightEntry.getValue(),
+                    mss.remainingOrderBy());
+
+            if (context.isRoot()) {
+                return consumingPlanner.plan(join, context);
+            }
+            return context.plannerContext().planSubRelation(join, context);
+        }
+
+        private List<OutputName> generateOutputNames(List<Symbol> outputs) {
```
Comment (dobe) [inline sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java]: use indexed fields

File: sql/src/main/java/io/crate/analyze/OrderBy.java [hunk 2]
```diff
@@ -123,4 +146,8 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeOptionalBoolean(nullFirst);
         }
     }
+
+    public OrderBy replaceSymbols(Function<? super Symbol, Symbol>replaceFunction) {
```
Comment (dobe) [inline sql/src/main/java/io/crate/analyze/OrderBy.java]: this is actually a copy, right? probably rename the method?

File: sql/src/main/java/io/crate/analyze/QuerySpec.java [hunk 3]
```diff
@@ -185,4 +185,71 @@ public int castOutputs(Iterator<DataType> types) {
...
+        for (Symbol output : outputs) {
+            newOutputs.add(replaceFunction.apply(output));
+        }
+        outputs = newOutputs;
+
+        if (where.hasQuery()) {
+            where = new WhereClause(replaceFunction.apply(where.query()), where.docKeys().orNull(), where.partitions());
+        }
+        if (orderBy.isPresent()) {
+            orderBy = Optional.of(orderBy.get().replaceSymbols(replaceFunction));
```
Comment (dobe) [inline sql/src/main/java/io/crate/analyze/QuerySpec.java]: this creates a copy, by intent?

File: sql/src/main/java/io/crate/analyze/symbol/Symbols.java [hunk 4]
```diff
@@ -91,6 +91,12 @@ public static boolean containsColumn(Iterable<? extends Symbol> symbols, ColumnI
         return false;
     }
 
+    public static boolean containsRelation(Symbol symbol, AnalyzedRelation relation) {
```
Comment (dobe) [inline sql/src/main/java/io/crate/analyze/symbol/Symbols.java]: is this unused?

File: sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+        private static final FieldToRelationColumnVisitor INSTANCE = new FieldToRelationColumnVisitor(true);
+
+        public FieldToRelationColumnVisitor(boolean inPlace) {
+            super(inPlace);
+        }
+
+        @Override
+        public Symbol visitField(Field field, FieldToRelationColumnCtx ctx) {
+            QualifiedName qualifiedName = ctx.relationToName.get(field.relation());
+            int idx = 0;
```
Comment (dobe) [inline sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java]: why not use field.index?

Comment (mfussenegger) [inline sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java]: The RelationColumn needs to point to the place inside the sub querySpec (left or right releation) and not to the place of the MSS querySpec

**Comment by seut** [comment]
ack

**Pull request merged**